workflow_zstack/paths.py

```
@@ -0,0 +1,47 @@
+from collections.abc import Sequence
+from typing import List
+import pathlib
+import datajoint as dj
+from element_interface.utils import find_full_path
+from element_session import session_with_id as session
+
+
+def get_volume_root_data_dir() -> List[str]:
+    """Return root directory for volumetric data in dj.config
+
+    Returns:
+        path (any): List of path(s) if available or None
+    """
+    vol_root_dirs = dj.config.get("custom", {}).get("volume_root_data_dir", None)
+    if not vol_root_dirs:
+        return None
+    elif not isinstance(vol_root_dirs, Sequence):
+        return list(vol_root_dirs)
+    else:
+        return pathlib.Path(vol_root_dirs)
+
+
+def get_volume_tif_file(scan_key):
+    """Retrieve the list of ScanImage files associated with a given Scan.
+
+    Args:
+        scan_key (dict): Primary key from Scan.
+
+    Returns:
+        path (list): Absolute path(s) of the scan files.
+
+    Raises:
+        FileNotFoundError: If the tiff file(s) are not found.
+    """
+    # Folder structure: root / subject / session / .tif (raw)    
+    sess_dir = find_full_path(
+        get_volume_root_data_dir(),
+        pathlib.Path((session.SessionDirectory & scan_key).fetch1("session_dir")),
+    )
+    
+    tiff_filepaths = [fp.as_posix() for fp in sess_dir.rglob("*.tif")][0]
```

```suggestion
    tiff_filepaths = [fp.as_posix() for fp in sess_dir.rglob("*.tif")]
```

---

Should we retrieve a single file or list?

---

We can retrieve a single file with
```
for file in sess_dir.glob("*.tif"):
    tiff_filepaths = file
```

The current code returns the file path in a list even if it's just one file so we'll either have to handle it here or in the make().

---

To clarify, it looks like the goal in the `make` is to find one file and load that single tiff file?  If so, then we should throw an error if there are multiple tiff files in the session directory, and only return a single file path.

---

I have updated the code here to assert that there is only one `tif` file in the list and this is the only file path that is returned.